modal/sandbox.py

```diff
@@ -193,11 +193,21 @@ async def _load(self: _Sandbox, resolver: Resolver, _existing_object_id: Optiona
     def _hydrate_metadata(self, handle_metadata: Optional[Message]):
         self._stdout = LogsReader(api_pb2.FILE_DESCRIPTOR_STDOUT, self.object_id, self._client)
         self._stderr = LogsReader(api_pb2.FILE_DESCRIPTOR_STDERR, self.object_id, self._client)
-        if handle_metadata is not None:
-            assert isinstance(handle_metadata, api_pb2.SandboxHandleMetadata)
-            self._result = handle_metadata.result
-        else:
-            self._result = None
+        self._result = None
+
+    @staticmethod
+    async def from_id(sandbox_id: str, client: Optional[_Client] = None) -> "_Sandbox":
+        """Construct a Sandbox from an id and look up the sandbox result."""
+        if client is None:
+            client = await _Client.from_env()
+
+        req = api_pb2.SandboxWaitRequest(sandbox_id=sandbox_id, timeout=0)
```

Comment:
@aksh-at sanity checking that this will work? doing this lets me remove the last use of `AppLookupObject`

These lookups are incredibly rare btw

Reply:
There's a _slight_ difference now. We'll set the `obj._result` to an empty `api_pb2.GenericResult()`, where it used to be `None`. But looking at the code, that shouldn't break anything.